source/dotnet/Library/AdaptiveCards.Rendering.Wpf/AdaptiveCardRenderer.cs

```diff
@@ -165,20 +166,29 @@ void Callback(object sender, AdaptiveActionEventArgs args)
 
         private async Task<RenderedAdaptiveCardImage> RenderCardToImageInternalAsync(AdaptiveCard card, int width, CancellationToken cancellationToken)
         {
-            var cardAssets = await LoadAssetsForCardAsync(card, cancellationToken);
+            RenderedAdaptiveCardImage renderCard = null;
 
-            var context = new AdaptiveRenderContext(null, null)
+            try
             {
-                CardAssets = cardAssets,
-                ResourceResolvers = ResourceResolvers,
-                ActionHandlers = ActionHandlers,
-                Config = HostConfig ?? new AdaptiveHostConfig(),
-                Resources = Resources,
-                ElementRenderers = ElementRenderers
-            };
+                var cardAssets = await LoadAssetsForCardAsync(card, cancellationToken);
 
-            var stream = context.Render(card).RenderToImage(width);
-            var renderCard = new RenderedAdaptiveCardImage(stream, card, context.Warnings);
+                var context = new AdaptiveRenderContext(null, null)
+                {
+                    CardAssets = cardAssets,
+                    ResourceResolvers = ResourceResolvers,
+                    ActionHandlers = ActionHandlers,
+                    Config = HostConfig ?? new AdaptiveHostConfig(),
+                    Resources = Resources,
+                    ElementRenderers = ElementRenderers
+                };
+
+                var stream = context.Render(card).RenderToImage(width);
+                renderCard = new RenderedAdaptiveCardImage(stream, card, context.Warnings);
+            }
+            catch(Exception e)
+            {
+                Debug.WriteLine($"RENDER Failed. {e.Message}");
```

Comment:
Only thing that seems weird is swallowing the exception here. Should we throw it? Or do we need to use a TaskCompletionSource and .SetException() ?

Reply:
Throwing the exception is what causes the unhandled exception to take down the process. We definitely don't want to do that.